models/app.model.js

```diff
@@ -49,50 +49,63 @@ exports.findAllCommentsFromArticle = (id) => {
     return db.query(`
         SELECT * FROM articles
         WHERE article_id = $1;
-    `,[id])
-    .then(( {rows} ) => {
-        if (!rows.length) {
-            return Promise.reject({ status: 404, msg: 'Article does not exist' })
-        }
-        return rows
-    })
-    .then((rows) => {
-    return db.query(`
+    `, [id])
+        .then(({ rows }) => {
+            if (!rows.length) {
+                return Promise.reject({ status: 404, msg: 'Article does not exist' })
+            }
+            return rows
+        })
+        .then((rows) => {
+            return db.query(`
         SELECT * FROM comments
         WHERE article_id = $1;
-        `,[id])
-    })
-    .then(({rows}) => {
-        return rows
-    })
+        `, [id])
+        })
+        .then(({ rows }) => {
+            return rows
+        })
 }
 
 exports.addCommentToArticle = (comment, id) => {
-        return db.query(`
+    return db.query(`
         INSERT INTO comments (body, author, article_id, votes, created_at)
         VALUES ($1, $2, $3, $4, $5) RETURNING *;
-        `, [comment.body, comment.author, comment.article_id, comment.votes, comment.created_at])  
-    .then(( {rows} ) => {
-        if (id != comment.article_id) {
-            return Promise.reject({ status: 404, msg: 'Article does not exist' })
-        }
-        if (!rows.length) {
-            return Promise.reject({ status: 404, msg: 'Article does not exist' })
-        }
+        `, [comment.body, comment.author, comment.article_id, comment.votes, comment.created_at])
+        .then(({ rows }) => {
+            if (id != comment.article_id) {
+                return Promise.reject({ status: 404, msg: 'Article does not exist' })
+            }
+            if (!rows.length) {
+                return Promise.reject({ status: 404, msg: 'Article does not exist' })
+            }
             return rows
         })
 };
 
-exports.updateVotes = (votes,id) => {
+exports.updateVotes = (votes, id) => {
     return db.query(`
     UPDATE articles
     SET votes = votes + $1
     WHERE article_id = $2 RETURNING *
-    `, [votes.inc_votes, id])  
-.then(( {rows} ) => {
-    if (!rows.length) {
-        return Promise.reject({ status: 404, msg: 'Article does not exist' })
-    }
-        return rows
-    })
-};
\ No newline at end of file
+    `, [votes.inc_votes, id])
+        .then(({ rows }) => {
+            if (!rows.length) {
+                return Promise.reject({ status: 404, msg: 'Article does not exist' })
+            }
+            return rows
+        })
+};
+
+exports.filterArticlesByTopic = (query) => {
+    return db.query(`
+    SELECT * FROM articles
+    WHERE topic = $1;
+    `, [query])
+        .then(({ rows }) => {
+            if (!rows.length) {
+                return Promise.reject({ status: 404, msg: 'Topic does not exist' })
```

Comment:
There is a situation we are not thinking about here - what if a topic existed...but there are no articles for it yet? This should return a 200 status code and an empty array